File: backend/src/NLC.Core/Entities/JobTypeConfig.cs
using NLC.Core.Enums;

namespace NLC.Core.Entities;

public class JobTypeConfig
{
    public Guid     Id               { get; set; } = Guid.NewGuid();
    public string   Name             { get; set; } = default!;         // e.g. "INBOUND"
    public string[] Phases           { get; set; } = [];               // ordered phase names
    public bool     VasOptional      { get; set; }
    public string?  GrnTriggerPhase  { get; set; }
    public string?  ErpPushPhase     { get; set; }
    public bool     IsActive         { get; set; } = true;

    public ICollection<JobCard> JobCards { get; set; } = [];
}

File: backend/src/main/java/com/nlc/domain/entity/JobTypeConfig.java
package com.nlc.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import java.util.UUID;

@Entity
@Table(name = "job_type_configs")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class JobTypeConfig {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(name = "phases", columnDefinition = "text[]")
    @JdbcTypeCode(SqlTypes.ARRAY)
    @Builder.Default
    private String[] phases = new String[0];

    @Column(name = "vas_optional", nullable = false)
    private boolean vasOptional;

    @Column(name = "grn_trigger_phase")
    private String grnTriggerPhase;

    @Column(name = "erp_push_phase")
    private String erpPushPhase;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private boolean isActive = true;
}
